trapdata/api/datasets.py

```
@@ -87,3 +100,280 @@ def __getitem__(self, idx):
+            # Use dedicated session without auth for external images
+            response = self.image_fetch_session.get(image_url, timeout=30)
+            response.raise_for_status()
+            image = Image.open(BytesIO(response.content))
+
+            # Convert to RGB if necessary
+            if image.mode != "RGB":
+                image = image.convert("RGB")
+
+            # Apply transforms
+            image_tensor = self.image_transforms(image)
+            return image_tensor
```

Ruff (0.14.14)

Consider moving this statement to an `else` block

(TRY300)